### ISSUE TITLE
Allow users to specify which services they want deployed through the manual SFDX deployment

### ISSUE THREAD

**Issue by lpatino10** [watson-developer-cloud/salesforce-sdk#27]
Right now, users can specify which services they want deployed, as opposed to always deploying every supported Watson service. However, this is only possible currently with the Ant deployment. This should be supported using SFDX as well.

**Comment by maniax89** [comment]
One thought I had for this is potentially using the `sfdx-project.json` coupled with moving the services around to their own packages.

```
.
├── _core
│   └── classes
│       ├── IBMWatsonClient.cls
│       ├── ...ommitted for brevity...
│       └── IBMWatsonValidator.cls-meta.xml
├── conversation
│   └── classes
│       ├── IBMConversationV1.cls
│       ├── IBMConversationV1.cls-meta.xml
│       ├── IBMConversationV1Models.cls
│       ├── IBMConversationV1Models.cls-meta.xml
│       ├── IBMConversationV1Test.cls
│       └── IBMConversationV1Test.cls-meta.xml
├── discovery
│   └── classes
│       ├── IBMDiscoveryV1.cls
│       ├── IBMDiscoveryV1.cls-meta.xml
│       ├── IBMDiscoveryV1Models.cls
│       ├── IBMDiscoveryV1Models.cls-meta.xml
│       ├── IBMDiscoveryV1Test.cls
│       └── IBMDiscoveryV1Test.cls-meta.xml
└── test
```

then maybe we have a main `sfdx-project.json` that looks like this:

```
{
  "packageDirectories": [
    {
      "path": "force-app/main/_core",
      "default": true
    },
    {
      "path": "force-app/main/discovery"
    },
    {
      "path": "force-app/main/conversation"
    },
  ],
  "namespace": "",
  "sfdcLoginUrl": "https://login.salesforce.com",
  "sourceApiVersion": "41.0"
}
```

then potentially a sfdx source push wrapper script that takes in an optional parameter (`--service-name discovery` maybe?) that either munges the JSON and replaces the `sfdx-project.json` or copies an existing template

**Comment by maniax89** [comment]
a side note to all of this - there is a `package2` api in beta right now that looks promising as well https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_dev2gp_workflow.htm

**Comment by magdielhf** [comment]
[This](https://github.com/watson-developer-cloud/salesforce-sdk/blob/sfdx-service-pkg/install/sfdx-deploy-service.sh) script uses a similar approach to the deployment instructions we've written for the `ant` deployment tool.

The way I'm thinking it'll work is by maintaining a configuration file in JSON format that would look similar to this

```JSON
{
  "watsonServices": [
    {
      "name": "discovery",
      "apexClasses": [
        {
         "fileName": "IBMDiscoveryV1.cls",
         "metadata": "IBMDiscoveryV1.cls-meta.xml"
        },
        {
         "fileName": "IBMDiscoveryV1Models.cls",
         "metadata": "IBMDiscoveryV1Models.cls-meta.xml"
        },
        {
         "fileName": "IBMDiscoveryV1Models.cls",
         "metadata": "IBMDiscoveryV1Models.cls-meta.xml"
        }
      ]
    },
    {
      "name": "conversation",
      "apexClasses": [
        {
         "fileName": "IBMConversationV1.cls",
         "metadata": "IBMConversationV1.cls-meta.xml"
        },
        {
         "fileName": "IBMConversationV1Models.cls",
         "metadata": "IBMConversationV1Models.cls-meta.xml"
        },
        {
         "fileName": "IBMDiscoveryV1Test.cls",
         "metadata": "IBMDiscoveryV1Test.cls-meta.xml"
        }
      ]
    }
  ],
  "sourceApiVersion": "41.0"
}
`
The script would then generate the deployment package from the dependencies defined on this configuration and deploy to a target scratch org.

The current version assumes there is a default org already set, I'll be adding that option next.

**Comment by maniax89** [comment]
For what it's worth, I had a similar issue that I solved with the comment suggested in https://github.com/forcedotcom/LightningTestingService/issues/49#issuecomment-350845077

If the content is divided into separate directories, then a user can select using the provided sfdx commands, i.e.

`sfdx waw:project:pdir:delete -p force-app/conversation`

**Comment by magdielhf** [comment]
For consistency with the `ant` deployment script this version of the SFDX script does not require maintaining a folder structure.

**Comment by germanattanasio** [comment]
Fixed in  267fc4d
